python/numpy.h

```diff
@@ -80,6 +80,11 @@ template <class T, class View>
 bool memory_overlaps(const py::array_t<T> &data, const View &view) {
   const auto &buffer_info = data.request();
   const auto [data_begin, data_end] = memory_begin_end<T>(buffer_info);
+  const auto begin = view.begin();
+  const auto end = view.end();
+  if (begin == end) {
+    return false;
+  }
```

Comment:
I suppose your plan was to use `begin` and `end` below? As it is now I see little point in having named variables for thise?

Reply:
No, the point is that I need the check `begin == end` before dereferencing the iterators, because otherwise `*begin` is a segfault or something similar.

Reply:
Sure, but I was wondering why you defined `begin` and `end` rather than writing `if (view.begin() == view.end())`?

Reply:
So I don't have to call `begin()` and `end()` twice.

Reply:
Oh....